### ISSUE TITLE
Generate copy constructor on AssociativeArrayRecord

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#15107]
We're currently generating copy constructors only on `ArrayRecord` types, not `AssociativeArrayRecord` types. This leads to a compilation error, as call sites may assume the copy constructor is present: https://github.com/jOOQ/jOOQ/issues/15097

While https://github.com/jOOQ/jOOQ/issues/15097 fixes the compilation error also in patch releases, this issue actually adds the copy constructor in new minor releases.

**Comment by mdaepp** [comment]
Some additonal remarks:

In my opinion, the copy constructor should look like this (for the example provided in #15096):

```
    /**
     * Create a new <code>TEST.PKG_15097.T_PERMISSIONS_TAB</code> record
     */
    public TPermissionsTabRecord(Map<? extends String, ? extends Short> map) {
        this();
        putAll(map);
    }
```

One could argue that an additional copy constructor is only necessary if pojos = true, but for `ArrayRecord` types, such a copy constructor is added regardless of the pojos setting. For reasons of consistency, the behavior should be the same here.

**Comment by lukaseder** [comment]
> One could argue that an additional copy constructor is only necessary if pojos = true

I don't think we need this restriction. It's useful irrespective of the `pojos` flag.
